### PR TITLE
user create controller for POST requests for /users

### PR DESCRIPTION
**Pull request by m4v15**
relates #48

### CODE REVIEW

**Review by mattlub — APPROVED** [review]

File: tests/controllers/users.test.js [hunk 1]
```diff
@@ -41,6 +41,43 @@ tape('test /users GET returns list of users', t => {
 
 // Tests for: POST /users
 
+tape('testing adding user using POST requests to /users route', t => {
+  supertest(server)
+    .post('/users')
+    .send(validUser1)
+    .expect(201)
```
Comment (des-des) [inline tests/controllers/users.test.js:48]: wikid

File: src/controllers/users.js [hunk 2]
```diff
@@ -24,7 +24,17 @@ userControllers.getById = (req, res) => {
...
   // receives json for user in body
   // adds to db
-  // status codes: 201 (created), 400 (bad request)
+  // status codes: 201 (created), 500 (server error)
+  const newUser = new User(req.body)
+  newUser.save()
+    .then(user => {
+      res.status(201).send(user)
+    })
+    .catch(err => {
```
Comment (des-des) [inline src/controllers/users.js:33]: This might be something to think about

http://thecodebarbarian.com/mongoose-error-handling

Comment (des-des) [reply src/controllers/users.js:33]: Maybe not now though

Comment (des-des) [reply src/controllers/users.js:33]: https://stackoverflow.com/questions/20021324/restful-api-node-js-mongoose-handling-errors

Comment (des-des) [reply src/controllers/users.js:33]: Anyway I think for now this is a bit of a rabbit hole..

File: src/controllers/users.js [hunk 3]
```diff
@@ -24,7 +24,17 @@ userControllers.getById = (req, res) => {
...
+      // Sending back 500 error, may need changing when we think about how we validate
+      const errorObj = { message: `Database error: ${err.message}` }
+      res.status(500).send(errorObj)
```
Comment (des-des) [inline src/controllers/users.js:36]: Okay, can leave this error code for now

**Review by des-des — APPROVED** [review]
Okay, Not going to hold stuff up too much,

I think we need to make custom error handling middleware but not here

**Comment by des-des** [comment]
@m4v15 @mattlub okay assign to me once you have seen my comments

**Comment by des-des** [comment]
I will merge

File: src/controllers/users.js (hunk 3 above)
Comment (des-des) [reply src/controllers/users.js:36]: Okay, comments now in #62

**Comment by des-des** [comment]
Put comments in #62

**Pull request merged**